app/components/Settings/scheduler/scheduler.jsx

```diff
@@ -276,30 +277,30 @@ class Scheduler extends Component {
                         headerRowHeight={32}
                     />
                 </Row>
-                <SchedulerPreview
-                    onCloseBtnClick={this.closePreview}
+
+                <CreateModal
+                    initialCode={this.props.initialCode}
+                    open={loggedIn && this.state.createModalOpen}
+                    onClickAway={this.closeCreateModal}
+                    onSubmit={this.createQuery}
+                    dialect={this.props.dialect}
+                />
+                <PromptLoginModal
+                    open={!loggedIn && this.state.createModalOpen}
+                    onClickAway={this.closeCreateModal}
+                    onSubmit={this.props.openLogin}
+                />
+
+                <PreviewModal
                     onClickAway={this.closePreview}
                     query={this.state.selectedQuery}
+                    onSave={this.handleUpdate}
```

Comment:
I've just noticed while testing. We need to disable `onSave` and `onDelete` when `!loggedIn`.